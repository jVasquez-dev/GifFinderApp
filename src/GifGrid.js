import React  from 'react'
import PropTypes from 'prop-types';

import { useFetchGifs } from './hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data:images, loading } = useFetchGifs( category );
    console.log(images)
   
    return (
        <>
            <h3>{ category }</h3>
         
            <div className="card-grid">
                    { loading && <p className="animate__animated animate__flash">Loading...</p>}
                    { 
                        images.map( ( img ) => (
                            <GifGridItem 
                            key = {img.id}
                            { ...img }
                            />
                        ))
                    }
                
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}