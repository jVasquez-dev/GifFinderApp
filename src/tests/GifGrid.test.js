import React, { PureComponent } from 'react'
import { shallow } from "enzyme";

import { GifGrid } from '../GifGrid'
import { useFetchGifs } from '../hooks/useFetchGifs';
jest.mock( '../hooks/useFetchGifs' )

describe('<GifGrid /> tests', () => {

    test('should render correctly', () => {
    
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const category = 'Naruto';
        const wrapper = shallow( <GifGrid category={ category }/>);
        
        expect( wrapper ).toMatchSnapshot()
    })

    test('should render correctly', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/folder/new.jpg',
            title: 'image'
        },
        {
            id: 'ABCD',
            url: 'https://localhost/folder/newa.jpg',
            title: 'image'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={ 'hola' }/>);
        
        expect( wrapper ).toMatchSnapshot()
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );

    })
    

  
})