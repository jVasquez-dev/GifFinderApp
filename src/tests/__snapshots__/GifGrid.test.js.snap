// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid /> tests should render correctly 1`] = `
<Fragment>
  <h3>
    Naruto
  </h3>
  <div
    className="card-grid"
  >
    <p
      className="animate__animated animate__flash"
    >
      Loading...
    </p>
  </div>
</Fragment>
`;

exports[`<GifGrid /> tests should render correctly 2`] = `
<Fragment>
  <h3>
    hola
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="image"
      url="https://localhost/folder/new.jpg"
    />
    <GifGridItem
      id="ABCD"
      key="ABCD"
      title="image"
      url="https://localhost/folder/newa.jpg"
    />
  </div>
</Fragment>
`;
